src/lib/precondition.ts

```diff
@@ -309,6 +326,10 @@ function assertPreconditionInvariants(accountUpdate: AccountUpdate) {
 
     // we accessed a precondition field but not constrained it explicitly - throw an error
     let hasAssertBetween = isRangeCondition(precondition);
+    // TODO: maybe the "validWhile" should be changed to "globalSlot" in a more global way?
```

Comment:
+1. If we're in a hurry it could be deferred I guess but this is the kind of debt that's hard to prioritise at a later date so I'm in favour of doing it now.